### ISSUE TITLE
twisted/test/test_protocols.py fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8717]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8717|
|Type|enhancement|
|Created|2016-08-02 11:26:59Z|
|Branch|https://github.com/twisted/twisted/tree/8717-rodrigc-protocols-py3|

If I try to run:

tox -e py35-tests twisted/test/test_protocols.py

it fails.

<details><summary>Searchable metadata</summary>

```
trac-id__8717 8717
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8717_rodrigc_protocols_py3 8717-rodrigc-protocols-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1470137219616438 1470137219616438
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/474

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_2"></a>|@acabhishek942 commented|
|-|-|

changes looks good.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset a92af71eb405a57be477d6970ed91e256c13fbbb
```
#!CommitTicketReference repository="" revision="a92af71eb405a57be477d6970ed91e256c13fbbb"
Merge pull request #474 from twisted/8717-rodrigc-protocols-py3

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8717

Fix twisted.test.test_protocols on Python 3
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
